### PR TITLE
Added create account to app

### PR DESCRIPTION
**Pull request by chandlerroth**
- Added username and password sign up flow using Firebase Auth

### CODE REVIEW

**Comment by kristencoy** [comment]
Looks great!

**Comment by chandlerroth** [comment]
Awesome @kristencoy! Thanks for the review :)

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,46 @@
...
+  const isEmailValid = (email) => {
+    return emailValidator.validate(email)
+  }
+
+  const createAccount = async (email, password) => {
+    if (isPasswordValid(password) && isEmailValid(email)) {
+      const auth = getAuth();
+
+      try {
+        const user = await createUserWithEmailAndPassword(auth, email, password)
```
Comment (ATMartin) [inline src/App.js:23]: nitpick: import function as createUser to be less verbose

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
+      }
+    }
+  }
 
   return (
-    <p>{query?.data?.hello}</p>
+    <div class="container">
+      <h2>Sign up</h2>
+      <label>
+        <input type="text" placeholder="Email" onChange={(event) => setEmail(event.target.value)} value={email} />
```
Comment (ATMartin) [inline src/App.js:35]: nitpick: Could we identify this input with a unique ID like `id="email-input"`

File: src/App.js [hunk 3]
```diff
@@ -1,15 +1,46 @@
...
+        Must contain valid email.
+      </label>
+      <label>
+        <input type="password" placeholder="Password" onChange={(event) => setPassword(event.target.value)} value={password} />
```
Comment (ATMartin) [inline src/App.js:39]: nitpick: Could we identify this input with a unique ID like `id="password-input"`

File: src/App.js [hunk 4]
```diff
@@ -1,15 +1,46 @@
...
 
   return (
-    <p>{query?.data?.hello}</p>
+    <div class="container">
+      <h2>Sign up</h2>
+      <label>
+        <input type="text" placeholder="Email" onChange={(event) => setEmail(event.target.value)} value={email} />
+        Must contain valid email.
+      </label>
+      <label>
```
Comment (ATMartin) [inline src/App.js:38]: blocker: Label tags need a `for` attribute to reference the input `id` for accessibility.

File: src/App.js [hunk 5]
```diff
@@ -1,15 +1,46 @@
 import './App.css';
-import { useQuery } from '@tanstack/react-query';
+import { getAuth, createUserWithEmailAndPassword } from "firebase/auth"
+import { useState } from 'react';
```
Comment (ATMartin) [inline src/App.js:3]: nitpick: Let's stay consistent and not use semicolons here

File: src/App.js [hunk 6]
```diff
@@ -1,15 +1,46 @@
...
+    return password.length >= 8
+  }
+
+  const isEmailValid = (email) => {
+    return emailValidator.validate(email)
+  }
+
+  const createAccount = async (email, password) => {
+    if (isPasswordValid(password) && isEmailValid(email)) {
+      const auth = getAuth();
```
Comment (ATMartin) [inline src/App.js:20]: nitpick: More semicolons?! Come on! 😂

**Pull request merged**